src/python/install.sh

```diff
@@ -264,7 +265,11 @@ install_from_source() {
     if [ "${OPTIMIZE_BUILD_FROM_SOURCE}" = "true" ]; then
         config_args="--enable-optimizations"
     fi
-    ./configure --prefix="${INSTALL_PATH}" --with-ensurepip=install ${config_args}
+    local shared_libs=""
+    if [ "${SHARED_LIBS}" = "true" ]; then
+        shared_libs="--enable-shared"
+    fi
+    ./configure --prefix="${INSTALL_PATH}" --with-ensurepip=install ${config_args} ${shared_libs}
```

Comment:
```suggestion
    ./configure --prefix="${INSTALL_PATH}" --with-ensurepip=install ${config_args} --enable-shared
```

Perhaps this is a feature we want by default and no configuration option is needed.

Reply:
Do you happen to know the performance consequences of adding `--enable-shared`?  I suspect that this would be to have as a default (perhaps inverting the option to let us disable) as long as it doesn't significantly increase the (already long) compile time

Reply:
Unfortunately I do not, the 'default' Python Docker compilation includes other flags that might improve performance however?

https://github.com/docker-library/python/blob/c007698b9f91bedba5e6300d9394c10acd194dcb/3.11/bookworm/Dockerfile#L44-L53